app/models/user.rb

```diff
@@ -138,13 +141,40 @@ def active_for_authentication?
   def serving_transition_aged_youth?
     actively_assigned_and_active_cases.is_transitioned.any?
   end
+
+  def record_previous_email
+    if email_changed? && !old_emails.include?(email_was)
+      old_emails.push(email_was)
+    end
+  end
+
+  def filter_old_emails(previous_email)
```

Comment:
my only nitpick is that i think this method name would be clearer if you added a bang at the end of it.  `filter_old_emails!` 

seeing how it's used in the files above, i wasn't sure if it was supposed to modify the objects or return them after filtering.